discovery.md

```diff
@@ -101,18 +101,295 @@ be considered an event subscriber and a target user of this API.
 
 ## API Specification
 
-This document structure focuses on discovery where each document has a two part
-key of `provider` and `type`. This takes a denormalized view where the composite
-key of documents is the `provider` and `type`. Provider is defined as an
-arbitrary human readable string and `type` is the CloudEvent `type` attribute.
+This API is specified as a REST API with well defined entities and relationships
+between those entities.
 
-### Provider entity Attributes
+All responses will have a content-type of `application/json` and are either
```

Comment:
I wonder if it would read better to put the definition of the resource model first, and then talk about the shape of the API.  This way the people get familiar with the entities and then this API section becomes more of a syntax definitional thing.